### DISCUSSION TITLE
Does core function have a caching mechanism ?

### DISCUSSION THREAD

**Discussion by 0xbuidler** [wevm/wagmi#2226] [Q&A]
Does functions like `readContract` from `@wagmi/core` has the same caching mechanism as the hook functions ? I cannot see any stale or cache params.

**Comment by jxom** [comment]
Not yet. But we will be exposing our Query Keys/Functions in `@wagmi/core` in the future, so you can take advantage of the caching mechanisms. See https://github.com/wagmi-dev/wagmi/discussions/2207 & search for **Export Query Keys/Functions**.
